### PR TITLE
[🚩esl-share]: final follow up

### PR DESCRIPTION
**Pull request by dshovchko**
Closes: #2012 

Contains next points:
 - fixed deduplication behavior on selecting buttons from config
 - fixed visual issue with displaying a huge list of share buttons in a popup
 - added ready state attribute and ready state event
 - revised observed attributes
 - retired esl-share and renamed esl-share-popup-trigger into esl-share
 - added ExportNS using
 - added support types for correct use in Typescript

### CODE REVIEW

File: src/modules/esl-share/test/esl-share-action-registry.test.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    afterEach(() => {
+      ESLShareActionRegistry.instance['actionsMap'].clear();
+    });
+
+    test('should throw an error when register invalid action', () => {
+      const {instance} = ESLShareActionRegistry;
+      expect(() => instance.register({} as any)).toThrowError();
+    });
+
+    test('should registers action', () => {
```
Comment (fshovchko) [inline src/modules/esl-share/test/esl-share-action-registry.test.ts:23]:
```suggestion
    test('should register action', () => {
```

**Review by fshovchko — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

File: src/modules/esl-share/core/esl-share.shape.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
-import type {ESLBaseElementShape} from '../../esl-base-element/core/esl-base-element.shape';
+import type {ESLTriggerTagShape} from '../../esl-trigger/core/esl-trigger.shape';
 import type {ESLShare} from './esl-share';
 
 /**
- * Tag declaration interface of ESL Share element
+ * Tag declaration interface of ESLSharePopupTrigger element
```
Comment (NastaLeo) [inline src/modules/esl-share/core/esl-share.shape.ts:5]: ESLShare?

File: src/modules/esl-share/core/esl-share.ts [hunk 3]
```diff
@@ -1,100 +1,132 @@
...
+    if (!this.isTargetActive) return;
+    this.$target?.hide();
   }
 
-  /** Appends share list to the share component. */
-  protected appendList(): void {
-    const $list = ESLShareList.create();
-    $list.list = this.list;
-    this.appendChild($list);
+  /** Update `$target` Toggleable  from `target` selector */
```
Comment (abarmina) [inline src/modules/esl-share/core/esl-share.ts:95]: extra cosmetic
```suggestion
  /** Updates `$target` Toggleable  from `target` selector */
```

Comment (NastaLeo) [inline src/modules/esl-share/core/esl-share.ts:95]:
```suggestion
  /** Updates `$target` Toggleable  from `target` selector */
```

**Review by NastaLeo — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

**Pull request merged**